cpp/src/parquet/column_reader.cc

```diff
@@ -1045,6 +1052,7 @@ class TypedColumnReaderImpl : public TypedColumnReader<DType>,
 
     // Not present for non-repeated fields
     if (this->max_rep_level_ > 0 && rep_levels != nullptr) {
+      DCHECK_GT(this->max_rep_level_, 0);
       int64_t num_rep_levels = this->ReadRepetitionLevels(batch_size, rep_levels);
       if (def_levels != nullptr && *num_def_levels != num_rep_levels) {
```

Comment:
Shouldn't this be `num_rep_levels != batch_size` instead?

Reply:
Actually it can 🤔